### PR TITLE
Add tests to increase code coverage

### PR DESCRIPTION
**Pull request by xTEddie**
image

### CODE REVIEW

File: __tests__/OmnichannelChatSDK.spec.ts [hunk 1]
```diff
@@ -116,7 +121,94 @@ describe('Omnichannel Chat SDK', () => {
...
+            try {
+                await chatSDK.startChat();
+            } catch (error) {
+                expect(console.error).toHaveBeenCalled();
+            }
+
+            expect(chatSDK.OCClient.sessionInit).toHaveBeenCalledTimes(1);
+            expect(chatSDK.IC3Client.initialize).toHaveBeenCalledTimes(1);
+            expect(chatSDK.IC3Client.joinConversation).toHaveBeenCalledTimes(1);
+        });
```
Comment (shurane) [inline __tests__/OmnichannelChatSDK.spec.ts:209]: Should there be a test where `OCClient.sessionInit()`, `IC3Client.initialize()`, and `IC3Client.joinConversation()` all succeed?

One worry I have is these tests are all very similar to each other. I guess I'm not sure how much value these tests are bringing. What do you think?

File: __tests__/OmnichannelChatSDK.spec.ts [hunk 2]
```diff
@@ -237,6 +342,83 @@ describe('Omnichannel Chat SDK', () => {
...
+
+            await chatSDK.startChat();
+
+            const chatContext = await chatSDK.getCurrentLiveChatContext();
+
+            expect(Object.keys(chatContext).includes('chatToken')).toBe(true);
+            expect(Object.keys(chatContext).includes('requestId')).toBe(true);
+        });
+
+        it('ChatSDK.getCurrentLiveChatContext() with empty chatToken should return an empty chat sesssion data', async () => {
```
Comment (shurane) [inline __tests__/OmnichannelChatSDK.spec.ts:372]: typo, should be "chat session data" (2 's' instead of 3).

**Review by shurane — APPROVED** [review]

File: __tests__/OmnichannelChatSDK.spec.ts [hunk 3]
```diff
@@ -476,5 +994,80 @@ describe('Omnichannel Chat SDK', () => {
...
+            };
+
+            const sessionCloseOptionalParams = {
+                authenticatedUserToken: optionaParams.authenticatedUserToken
+            }
+
+            await chatSDK.endChat();
+            expect(chatSDK.OCClient.sessionClose).toHaveBeenCalledTimes(1);
+            expect(chatSDK.OCClient.sessionClose.mock.calls[0][1]).toMatchObject(sessionCloseOptionalParams);
+        });
```
Comment (shurane) [inline __tests__/OmnichannelChatSDK.spec.ts:1071]: I think the tests in OmniChannelChatSdk.test.ts is great and I'm okay if we just keep it like this so we can increase our code coverage numbers.

I have some comments though, maybe we can have a conversation about them.

There's a lot of boiler plate in each test in this file. I had some ideas regarding this.

1. What do you think of fewer tests (maybe even just one) that check the critical path of a successful conversation (i.e. startChat(), getMessages(), sendMessage(), downloadFile(), getLiveChatTranscript(),  emailLiveChatTranscript(), endChat()).

2. We could also have one that checks where various things fail (sendMessage() and downloadFile() when there's no internet connection, startChat() without a proper authenticatedUserToken, etc).

3. Wouldn't we also want a way to test registering and receiving the various events? I.e. `onNewMessage()`, `onAgentEndSession()`, `onTypingEvent()`.

File: src/validators/SDKConfigValidators.ts [hunk 4]
```diff
@@ -13,10 +13,12 @@ const defaultChatSDKConfig: IChatSDKConfig = {
  */
 const validateDataMaskingConfig = (dataMaskingConfig: IDataMaskingSDKConfig) => {
     if (typeof dataMaskingConfig.disable !== "boolean") {
+        // console.warn('Data masking config disable flag must be true/false');
```
[the remark this answers is not in the document]
Comment (xTEddie) [reply src/validators/SDKConfigValidators.ts]: I removed those comments

File: __tests__/validators/SDKConfigValidators.spec.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import validateSDKConfig, {defaultChatSDKConfig} from '../../src/validators/SDKConfigValidators';
+
+describe('SDKConfigValidators', () => {
+    it('Data masking SDK config should be set to default values if not set', () => {
+        const chatSDKConfig: any = {};
+        chatSDKConfig.dataMasking = {};
+
+        validateSDKConfig(chatSDKConfig);
+
+        expect(chatSDKConfig.dataMasking.disable).toBe(defaultChatSDKConfig.dataMasking.disable);
```
[the remark this answers is not in the document]
Comment (xTEddie) [reply __tests__/validators/SDKConfigValidators.spec.ts:10]: Objects in JavaScript are pass by reference, so `toBe` would not work, so `toStrictEqual` had to be used.

File: __tests__/OmnichannelChatSDK.spec.ts (hunk 1 above)
Comment (xTEddie) [reply __tests__/OmnichannelChatSDK.spec.ts:209]: There's already a test where all succeed on ``'ChatSDK.startchat() should start an OC chat'`` test case

File: __tests__/OmnichannelChatSDK.spec.ts (hunk 3 above)
Comment (xTEddie) [reply __tests__/OmnichannelChatSDK.spec.ts:1071]: 1. I agree that we should have fewer, but meaningful tests. Code coverage is not the best indicator of the test quality. We also have to maintain the tests, so having too many can have an impact in our development productivity. We'll be able to have better test quality once we'll get our e2e test harness.
2. Agree. These tests would be meaningful. It should be part of e2e tests.
3. It'd be a good idea to test registering and receiving events. We'll have to find a way on how to test them.

**Pull request merged**